includes/abstracts/abstract-wc-legacy-order.php

```diff
@@ -0,0 +1,621 @@
+		} elseif ( 'display_totals_ex_tax' === $key ) {
+			return 'excl' === get_option( 'woocommerce_tax_display_cart' );
+		} elseif ( 'display_cart_ex_tax' === $key ) {
+			return 'excl' === get_option( 'woocommerce_tax_display_cart' );
+		} elseif ( 'cart_discount' === $key ) {
+			return $this->get_discount();
+		} elseif ( 'cart_discount_tax' === $key ) {
+			return $this->get_discount_tax();
+		} elseif ( 'order_tax' === $key ) {
+			return $this->get_cart_tax();
+		} elseif ( 'order_shipping_tax' === $key ) {
+			return $this->get_shipping_tax();
+		} elseif ( 'order_shipping' === $key ) {
+			return $this->get_shipping_total();
+		} elseif ( 'order_total' === $key ) {
+			return $this->get_total();
+		} elseif ( 'order_type' === $key ) {
+			return $this->get_type();
+		} elseif ( 'order_currency' === $key ) {
+			return $this->get_currency();
+		} elseif ( 'order_version' === $key ) {
+			return $this->get_version();
+	 	} elseif ( is_callable( array( $this, "get_{$key}" ) ) ) {
+			return $this->{"get_{$key}"}();
+		} else {
+			return get_post_meta( $this->get_id(), '_' . $key, true );
+		}
+	}
+
+	/**
+	 * has_meta function for order items.
+	 *
+	 * @param string $order_item_id
+	 * @return array of meta data.
+	 */
+	public function has_meta( $order_item_id ) {
+		global $wpdb;
+
+		_deprecated_function( 'has_meta', '2.7', 'WC_Order_item::get_meta_data' );
+
+		return $wpdb->get_results( $wpdb->prepare( "SELECT meta_key, meta_value, meta_id, order_item_id
+			FROM {$wpdb->prefix}woocommerce_order_itemmeta WHERE order_item_id = %d
+			ORDER BY meta_id", absint( $order_item_id ) ), ARRAY_A );
+	}
+
+	/**
+	 * Display meta data belonging to an item.
+	 * @param  array $item
+	 */
+	public function display_item_meta( $item ) {
+		_deprecated_function( 'get_item_meta', '2.7', 'wc_display_item_meta' );
+		$product   = $item->get_product();
+		$item_meta = new WC_Order_Item_Meta( $item, $product );
+		$item_meta->display();
+	}
+
+	/**
+	 * Display download links for an order item.
+	 * @param  array $item
+	 */
+	public function display_item_downloads( $item ) {
+		$product   = $item->get_product();
+
+		if ( $product && $product->exists() && $product->is_downloadable() && $this->is_download_permitted() ) {
+			$download_files = $this->get_item_downloads( $item );
+			$i			  = 0;
+			$links		  = array();
+
+			foreach ( $download_files as $download_id => $file ) {
+				$i++;
+				$prefix  = count( $download_files ) > 1 ? sprintf( __( 'Download %d', 'woocommerce' ), $i ) : __( 'Download', 'woocommerce' );
+				$links[] = '<small class="download-url">' . $prefix . ': <a href="' . esc_url( $file['download_url'] ) . '" target="_blank">' . esc_html( $file['name'] ) . '</a></small>' . "\n";
+			}
+
+			echo '<br/>' . implode( '<br/>', $links );
+		}
+	}
+
+	/**
+	 * Get the Download URL.
+	 *
+	 * @param  int $product_id
+	 * @param  int $download_id
+	 * @return string
+	 */
+	public function get_download_url( $product_id, $download_id ) {
```

Comment:
Should this throw a deprecate notice?

I noticed there is no alternative available, and instead, `WC_Order_Item_Product::get_item_downloads()` duplicates the code here. Perhaps there should be a `WC_Order_Item_Product: get_download_url()` or similar method created to replace this one.